src/vs/workbench/contrib/notebook/common/notebookEditorModel.ts

```diff
@@ -100,12 +101,14 @@ export class SimpleNotebookEditorModel extends EditorModel implements INotebookE
 		return !SimpleNotebookEditorModel._isStoredFileWorkingCopy(this._workingCopy) && !!this._workingCopy?.hasAssociatedFilePath;
 	}
 
-	isReadonly(): boolean {
+	isReadonly(): boolean | IMarkdownString {
 		if (SimpleNotebookEditorModel._isStoredFileWorkingCopy(this._workingCopy)) {
 			return this._workingCopy?.isReadonly();
-		} else if (this._filesConfigurationService.isReadonly(this.resource)) {
-			return true;
 		} else {
+			const readonly = this._filesConfigurationService.isReadonly(this.resource);
```

Comment:
This could be simplified, no? Just always return `this._filesConfigurationService.isReadonly...`?